src/Workspaces/CSharp/Portable/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -340,5 +340,15 @@ internal override ITypeInferenceService GetTypeInferenceServiceWithoutWorkspace(
         {
             return new CSharpTypeInferenceService();
         }
+
+        /// <summary>
+        /// Is this a possible position for an await statement (`await using` or `await foreach`)?
+        /// </summary>
+        internal bool IsAwaitStatementContext(int position, CancellationToken cancellationToken)
+        {
+            var leftToken = this.SyntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            var targetToken = leftToken.GetPreviousTokenIfTouchingWord(position);
+            return targetToken.Kind() == SyntaxKind.AwaitKeyword && targetToken.GetPreviousToken().IsBeginningOfStatementContext();
```

Comment:
What happens if targetToken.GetPreviousToken isn't a token? Try this in interactive scenarios where "await" is the first token in the file?

Reply:
`GetPreviousToken` can return `default`. `IsBeginningOfStatementContext` can handle that.